frontend/src/Frontend/UI/Widgets.hs

```diff
@@ -913,19 +914,38 @@ uiAccountFixed sender = do
 mkChainTextAccounts
   :: (Reflex t, HasWallet model key t, HasNetwork model t)
   => model
+  -> Bool
   -> Dynamic t (Maybe ChainId)
   -> Dynamic t (Either Text (Map AccountName Text))
-mkChainTextAccounts m mChainId = runExceptT $ do
+mkChainTextAccounts m needsGas mChainId = runExceptT $ do
   netId <- lift $ m ^. network_selectedNetwork
+  -- Calculating the max gas from the network meta causes a causality loop
+  -- maxGasPriceBalance <- lift $ AccountBalance . calcMaxGas <$> m ^. network_meta
+
   chain <- ExceptT $ note "You must select a chain ID before choosing an account" <$> mChainId
   accountsOnNetwork <- ExceptT $ note "No accounts on current network" . Map.lookup netId . unAccountData <$> m ^. wallet_accounts
   let mkVanity n (AccountInfo _ chainMap)
-        | Map.member chain chainMap = Map.singleton n (unAccountName n)
+        | Just a <- Map.lookup chain chainMap
+        , not needsGas || (fromMaybe 0 (a ^? account_status . _AccountStatus_Exists . accountDetails_balance)) > 0
```

Comment:
This will now fail if the account isn't found, though that sounds more or less impossible?

Reply:
Actually seems to be an improvement, given the error message we show.